### PR TITLE
SITES-7941: add a decoupled frontend project

### PR DESCRIPTION
**Pull request by buuhuu**
## Description

With this change it is possible to create an AEM SPA project that leverages the Frontend Pipeline to serve all the frontend files instead of clientlibs.

To do this a `decoupled` frontend module option got introduced which:
- does not create a ui.frontend maven module anymore (same as none),
- generates all the content and components like the other SPA projects (angular and react), but
- additionally adds the necessary configurations to pre-enable the Frontend Pipeline to serve the frontend code

This may be used in combination with https://github.com/adobe/aem-react-spa as a starting point for a react project that is
- editable using the SPA editor
- deployable via the Frontend Pipeline

## Related Issue

SITES-7941
https://github.com/adobe/aem-react-spa/pull/1
https://github.com/adobe/aem-spa-project-core/pull/60

## Motivation and Context

Further drive adaption of the Frontend Pipeline, in particular for SPA project.

## How Has This Been Tested?

Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/xfpage/customheaderlibs.html [hunk 1]
```diff
@@ -31,8 +31,16 @@
 
 <sly data-sly-use.clientlib="/libs/granite/sightly/templates/clientlib.html"/>
 
+#if ( $frontentModule != "decoupled" ) 
 <sly data-sly-call="${symbol_dollar}{clientlib.all @ categories=['coralui3']}"/>
 <sly data-sly-call="${symbol_dollar}{clientlib.css @ categories=['${appId}.base']}"/>
+#end
+
+<sly data-sly-list="${page.htmlPageItems}">
```
Comment (vladbailescu) [inline src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/xfpage/customheaderlibs.html:39]: We need to also update `customfooterlibs.html`

**Review by vladbailescu — CHANGES_REQUESTED** [review]
I think we should also provide (minimal) some instructions on how to setup the decoupled SPA project, the referenced docs don't cover this at the moment.

File: src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/xfpage/customheaderlibs.html (hunk 1 above)
Comment (buuhuu) [reply src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appId__/components/xfpage/customheaderlibs.html:39]: done

**Comment by buuhuu** [comment]
> I think we should also provide (minimal) some instructions on how to setup the decoupled SPA project, the referenced docs don't cover this at the moment.

There are some instructions in [aem-react-spa](https://github.com/adobe/aem-react-spa/pull/1). I linked them in the Features section.

**Review by vladbailescu — APPROVED** [review]

**Pull request merged**